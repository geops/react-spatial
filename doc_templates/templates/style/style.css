<style>
html {
  margin:0px;
  padding:0px;
  height: 100%;
}

body {
  margin:0px;
  padding:0px;
  height: 100%;
}

iframe {
  position: relative;
  border: 0;
  width: 100%;
  height: calc(100% - 70px);
}

h1, h2 {
  font-weight: normal;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  display: inline-block;
  margin: 0 10px;
}

a {
  color: #42b983;
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

a.active {
  font-weight: bold;
}

#header {
  border-bottom: 2px solid #61849c;
  color: #61849c;
  height: 20px;
  padding: 20px;
  text-align: right;
}

#header a {
  color: #61849c;
  margin-right: 10px;
}

#header a:last-child {
    margin-right: 0;
}

#map {
  position: absolute;
  left: 0;
  top: 62px;
  bottom: 0;
  right: 0;
 }

#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
  min-height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
  height: 100%;
}

#brand {
  font-size: 20px;
  font-weight: bold;
  height: 60px;
  line-height: 60px;
  left: 30px;
  position: absolute;
  top: 0;
}

#promo {
  position: absolute;
  bottom: 50px;
  background-color: red;
  right: -65px;
  transform: rotate(-45deg);
  z-index: 1;
}

#promo-text {
  border: 2px solid white;
  color: white;
  font-size: 14px;
  font-family: Sans-Serif;
  font-weight: bold;
  margin: 2px;
  padding: 0 70px;
}

#promo a:hover {
  text-decoration: none;
}

#copyrightDiv {
  font-family: Arial;
  font-size: 12px;
  background-color: rgba(255, 255, 255, 0.7);
  box-shadow: 1px 2px 4px rgba(0,0,0,0.7);
  position: absolute;
  padding: 5px;
  bottom: 0;
  left:0;
  right: 0;
}

</style>
