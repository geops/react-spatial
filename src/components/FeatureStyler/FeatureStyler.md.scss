.tm-feature-styler-example {
  position: relative;

  .tm-ole {
    position: absolute;
    top: 10px;
    right: 10px;
    width: 45px;

    button {
      background: #fafafa;
      border: 0;
      padding: 5px;
      width: 45px;
      height: 45px;
      box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.2);

      &.active {
        box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.3);
        color: #5c5c5c;
        filter: brightness(90%);
      }
    }

    img {
      width: 35px;
      height: 35px;
    }
  }

  .tm-feature-styler-popup {
    position: absolute;
    top: 10px;
    left: 10px;
    bottom: 10px;
    background: white;
    padding: 10px;
    width: 200px;
    overflow-y: auto;

    > .tm-button {
      position: absolute;
      top: 0;
      right: 0;
    }
  }

  .tm-feature-styler {
    & > div > * {
      margin: 10px 0;
    }
  }

  .tm-modify-icon-size {
    margin: 20px 0;

    select {
      padding: 10px;
      background: white;
      width: 100%;
    }
  }

  .tm-modify-icons {
    margin: 5px 0;

    div[role='button'] {
      display: inline-block;
      padding: 2px;

      &.tm-selected {
        background-color: lightgray;
      }

      &:hover {
        background-color: gray;
      }
    }
  }
}
