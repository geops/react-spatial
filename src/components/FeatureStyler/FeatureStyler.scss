.tm-feature-styler {
  .tm-button-text-bold {
    font-weight: bold;
  }
}
