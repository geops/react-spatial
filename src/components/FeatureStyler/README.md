#

This demonstrates the use of FeatureStyler.

```jsx
import React from 'react';
import BasicMap from 'react-spatial/components/BasicMap';
import Layer from 'react-spatial/Layer';
import VectorLayer from 'react-spatial/VectorLayer';
import OLMap from 'ol/Map';
import Feature from 'ol/Feature';
import { Point, LineString } from 'ol/geom';
import VectorSource from 'ol/source/Vector';
import TileLayer from 'ol/layer/Tile';
import Select from 'ol/interaction/Select';
import { Style, Fill, Icon, Text, Stroke } from 'ol/style';
import OLE from 'react-spatial/components/OLE';
import OSM from 'ol/source/OSM';
import FeatureStyler from 'react-spatial/components/FeatureStyler';
import Button from 'react-spatial/components/Button';
import AddTextIcon from 'react-spatial/images/text.png';

class FeatureStylerExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedFeature: null,
    };
    this.select = this.select.bind(this);
    this.deselect = this.deselect.bind(this);

    this.map = new OLMap();

    this.defaultIconStyle = new Style({
      image: new Icon({
        src: 'images/marker.png',
        scale: 0.5,
        imgSize: [24, 24], // ie 11
      }),
    });

    this.defaultTextStyle = new Style({
      text: new Text({
        font: '16px arial',
        text: 'My custom text',
        fill: new Fill({
          color: [255, 0, 0],
        }),
        stroke: new Stroke({
          color: [255, 255, 255],
          width: 3,
        }),
        scale: 1.5,
        rotation: 0.5,
      }),
    });

    this.defaultLineStyle = new Style({
      stroke: new Stroke({
        color: 'red',
        lineDash: [10, 10],
        width: 3,
      }),
    });

    // Draw labels
    this.drawCustomsOptions = [
      {
        style: this.defaultTextStyle.clone(),
        image: AddTextIcon,
        onDrawEnd: evt => {
          evt.feature.setStyle(this.defaultTextStyle.clone());
        },
      },
    ];

    // Draw icons
    this.drawIconOptions = {
      style: this.defaultIconStyle.clone(),
      onDrawEnd: evt => {
        evt.feature.setStyle(this.defaultIconStyle.clone());
      },
    };

    // Draw dashed line
    this.drawLineOptions = {
      style: this.defaultLineStyle.clone(),
      onDrawEnd: evt => {
        evt.feature.setStyle(this.defaultLineStyle.clone());
      },
    };

    const text = new Feature(new Point([-8000000, 3000000]));
    text.setStyle(this.defaultTextStyle.clone());

    const icon = new Feature(new Point([8000000, 3000000]));
    icon.setStyle(this.defaultIconStyle.clone());

    const line = new Feature(new LineString([
      [-8000000, 3000000], [8000000, 3000000],
    ]));
    line.setStyle(this.defaultLineStyle.clone());

    this.layers = [
      new Layer({
        olLayer: new TileLayer({
          source: new OSM(),
        }),
      }),
      new VectorLayer({
        source: new VectorSource({
          features: [text, icon, line],
        }),
      }),
    ];
  }

  select(feature) {
    this.setState({
      selectedFeature: feature,
    });
  }

  deselect() {
    this.setState({
      selectedFeature: null,
    });
  }

  renderFeatureStyler() {
    const { selectedFeature } = this.state;

    // Modification of feature Style is only allowed if a feature has a style.
    if (!selectedFeature || !selectedFeature.getStyleFunction()) {
      return null;
    }

    return (
      <div className="tm-feature-styler-popup">
        <FeatureStyler feature={selectedFeature} />
      </div>
    );
  }

  render() {
    return (
      <div className="tm-feature-styler-example">
        <BasicMap map={this.map} layers={this.layers} />
        <OLE
          map={this.map}
          drawPoint={this.drawIconOptions}
          drawLineString={this.drawLineOptions}
          drawCustoms={this.drawCustomsOptions}
          modify={this.modifyOptions}
          layer={this.layers[1]}
          onSelect={this.select}
          onDeselect={this.deselect}
        />
        {this.renderFeatureStyler()}
      </div>
    );
  }
}

<FeatureStylerExample />;
```
