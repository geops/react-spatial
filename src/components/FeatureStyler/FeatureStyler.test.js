import 'jest-canvas-mock';
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Feature } from 'ol';
import { Fill, Stroke, Style, Icon, Text } from 'ol/style';
import FeatureStyler from '.';

configure({ adapter: new Adapter() });

const dashedLine = new Style({
  stroke: new Stroke({
    color: 'red',
    lineDash: [10, 10],
    width: 3,
  }),
});

// Text style with a custom style
const simpleText = new Style({
  text: new Text({
    font: '16px Helvetica',
    text: 'foo',
  }),
});

// Icon style with a custom style
const simpleIcon = new Style({
  image: new Icon({
    src: 'images/favicon.png',
    scale: 3,
  }),
});

document.execCommand = jest.fn();

describe('FeatureStyler', () => {
  describe('matches snapshot', () => {
    test('when no feature provided', () => {
      const component = renderer.create(<FeatureStyler />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('when feature has no styleFunction() defined', () => {
      const component = renderer.create(
        <FeatureStyler feature={new Feature()} />,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    describe('when feature has a styleFunction() which', () => {
      const featReturnsStyle = new Feature();
      featReturnsStyle.setStyle(() => {
        return simpleIcon;
      });

      const featReturnsArray = new Feature();
      // The call to getStyleFunction()() will return an array: [simpleIcon];
      featReturnsArray.setStyle(simpleIcon);

      const featReturns3Styles = new Feature();
      featReturns3Styles.setStyle([simpleIcon, simpleText, dashedLine]);

      const featReturnsEmptyArray = new Feature();
      featReturnsEmptyArray.setStyle(() => {
        return [];
      });

      [
        {
          descr: ' a style object',
          feature: featReturnsStyle,
        },
        {
          descr: ' an array of one style',
          feature: featReturnsArray,
        },
        {
          descr: ' an empty array',
          feature: featReturnsEmptyArray,
        },
        {
          descr: ' an array of 3 styles targeting the one at index 1',
          feature: featReturns3Styles,
          idx: 1,
        },
      ].forEach(s => {
        test(`returns ${s.descr}.`, () => {
          const component = renderer.create(
            <FeatureStyler feature={s.feature} styleIdx={s.idx} />,
          );
          const tree = component.toJSON();
          expect(tree).toMatchSnapshot();
        });
      });

      test(`returns a style with icon and text values selected from properties.`, () => {
        const completeStyleModififiable = new Style({
          image: new Icon({
            src: 'images/marker.png',
            scale: 0.5,
          }),
          text: new Text({
            font: '16px arial',
            text: 'bar',
            fill: new Fill({
              color: [255, 0, 0],
            }),
            stroke: new Stroke({
              color: [255, 255, 255],
              width: 3,
            }),
            scale: 1.5,
            rotation: 0.5,
          }),
          fill: new Fill({
            color: [255, 0, 0],
          }),
          stroke: new Stroke({
            color: [255, 255, 255],
            width: 3,
          }),
        });
        const feature = new Feature();
        feature.setStyle(completeStyleModififiable);
        const component = renderer.create(<FeatureStyler feature={feature} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      test(`returns a style with color values selected from properties.`, () => {
        const completeStyleModififiable = new Style({
          fill: new Fill({
            color: [255, 0, 0],
          }),
          stroke: new Stroke({
            color: [255, 255, 255],
            width: 3,
          }),
        });
        const feature = new Feature();
        feature.setStyle(completeStyleModififiable);
        const component = renderer.create(<FeatureStyler feature={feature} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      test(`returns a style with bold font.`, () => {
        const completeStyleModififiable = new Style({
          text: new Text({
            font: 'bold 16px Helvetica',
          }),
        });
        const feature = new Feature();
        feature.setStyle(completeStyleModififiable);
        const component = renderer.create(<FeatureStyler feature={feature} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('#componentDidUpdate()', () => {
    const feature = new Feature();
    feature.setStyle(simpleIcon);

    test('updates content when feature changes', () => {
      const feature2 = new Feature();
      feature2.setStyle(simpleIcon);
      const wrapper = shallow(<FeatureStyler feature={feature} />);
      const inst = wrapper.instance();
      const spy = jest.spyOn(inst, 'updateContent');
      wrapper.setProps({
        feature: feature2,
      });
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith();
    });

    test("doesn't update content when new feature has no styleFunction", () => {
      const feature2 = new Feature();
      const wrapper = shallow(<FeatureStyler feature={feature} />);
      const inst = wrapper.instance();
      const spy = jest.spyOn(inst, 'updateContent');
      wrapper.setProps({
        feature: feature2,
      });
      expect(spy).toHaveBeenCalledTimes(0);
    });

    [
      'font',
      'name',
      'description',
      'color',
      {
        name: 'icon',
        value: { id: 'a', url: 'a' },
        value2: { id: 'a', url: 'a' },
      },
      {
        name: 'iconSize',
        value: { label: 'a', value: [0, 0], scale: 1 },
        value2: { label: 'a', value: [0, 0], scale: 1 },
      },
      'textColor',
      'textSize',
      'textRotation',
    ].forEach(p => {
      const name = typeof p === 'string' ? p : p.name;
      const value = typeof p === 'string' ? p : p.value;
      const value2 = typeof p === 'string' ? `${p}2` : p.value2;
      test(`apply the new style when state's property ${name} changes`, () => {
        const feature2 = new Feature();
        feature2.setStyle(simpleIcon);
        const wrapper = shallow(<FeatureStyler feature={feature} />);
        const inst = wrapper.instance();
        const spy = jest.spyOn(inst, 'applyStyle');
        const state = {};
        state[name] = value;
        inst.setState(state);
        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith();

        state[name] = value2;
        inst.setState(state);
        expect(spy).toHaveBeenCalledTimes(2);
        expect(spy).toHaveBeenCalledWith();
      });
    });
  });

  describe('#toggleFontBold()', () => {
    test('returns a bold font', () => {
      const font = FeatureStyler.toggleFontBold('16px Helvetica');
      const substring = 'bold';
      expect(font.includes(substring)).toBe(true);
    });

    test('returns a not bold font', () => {
      const font = FeatureStyler.toggleFontBold('bold 16px Helvetica');
      const substring = 'bold';
      expect(font.includes(substring)).toBe(false);
    });
  });

  describe('#convertDegRotation()', () => {
    test('returns a correct ol numeric rotation converted to deg', () => {
      const rotation = FeatureStyler.convertDegRotation(3.141592653589793);
      expect(rotation).toBe('180');
    });

    test('limits too high ol numeric rotation to 360 deg', () => {
      const rotation = FeatureStyler.convertDegRotation(99);
      expect(rotation).toBe('360');
    });

    test('limits negative ol numeric rotation to 0 deg', () => {
      const rotation = FeatureStyler.convertDegRotation(-99);
      expect(rotation).toBe('0');
    });
  });

  describe('when a fill style is modified', () => {
    test("set state's color property onClick of a color", () => {
      const feature = new Feature();
      feature.setStyle(
        new Style({
          fill: new Fill({
            color: [255, 0, 0],
          }),
          stroke: new Stroke({
            color: [255, 255, 255],
            width: 3,
          }),
        }),
      );
      const wrapper = mount(<FeatureStyler feature={feature} />);
      expect(wrapper.state().lineColor).not.toEqual(
        FeatureStyler.defaultProps.colors[1],
      );
      wrapper
        .find('.tm-color-picker')
        .at(0)
        .find('Button')
        .at(0)
        .simulate('click');
      wrapper
        .find('.github-picker')
        .find('span div span div')
        .at(1)
        .simulate('click');
      expect(wrapper.state().lineColor).toEqual(
        FeatureStyler.defaultProps.lineColors[1],
      );
    });
  });

  describe('when a text style is modified', () => {
    let wrapper;
    const feature = new Feature();
    feature.setStyle(simpleText);

    beforeEach(() => {
      wrapper = mount(<FeatureStyler feature={feature} />);
    });

    test("set state's text property on input change", () => {
      expect(wrapper.state().name).toEqual('foo');
      wrapper.find('textarea').simulate('change', { target: { value: 'bar' } });
      expect(wrapper.state().name).toEqual('bar');
    });

    test('toggle text boldness on click', () => {
      expect(wrapper.state().font).toEqual('16px Helvetica');
      const btn = wrapper
        .find('.tm-modify-text-font')
        .find('Button')
        .at(0);
      btn.simulate('click');
      expect(wrapper.state().font).toEqual('bold 16px Helvetica');
      btn.simulate('click');
      expect(wrapper.state().font).toEqual('16px Helvetica');
    });

    test("set state's text color property onClick of a color", () => {
      expect(wrapper.state().textColor).not.toEqual(
        FeatureStyler.defaultProps.colors[1],
      );
      wrapper
        .find('.tm-modify-text-color')
        .find('.tm-color-picker')
        .at(1)
        .find('Button')
        .at(0)
        .simulate('click');
      wrapper
        .find('.tm-modify-text-color')
        .find('.github-picker')
        .find('span div span div')
        .at(1)
        .simulate('click');
      expect(wrapper.state().textColor).toEqual(
        FeatureStyler.defaultProps.colors[1],
      );
    });

    test("set state's text size property on select", () => {
      expect(wrapper.state().textSize).not.toEqual(
        FeatureStyler.defaultProps.textSizes[1],
      );
      wrapper.find('Select').simulate('change', { target: { value: 2 } });
      expect(wrapper.state().textSize).toEqual(
        FeatureStyler.defaultProps.textSizes[2],
      );
    });

    test("set state's text rotation property on change input number", () => {
      expect(wrapper.state().textRotation).not.toEqual(3.141592653589793);
      wrapper
        .find('input')
        .at(0)
        .simulate('change', { target: { value: 180 } });
      expect(wrapper.state().textRotation).toEqual(3.141592653589793);
    });

    test("set state's text rotation property on change input range", () => {
      expect(wrapper.state().textRotation).not.toEqual(0.40142572795869574);
      wrapper
        .find('input')
        .at(1)
        .simulate('change', { target: { value: 23 } });
      expect(wrapper.state().textRotation).toEqual(0.40142572795869574);
    });
  });

  describe('when a icon style is modified', () => {
    let wrapper;
    const feature = new Feature();
    feature.setStyle(simpleIcon);

    beforeEach(() => {
      wrapper = mount(<FeatureStyler feature={feature} />);
    });

    test("set state's icon property onClick of a color", () => {
      const targetIcon = FeatureStyler.defaultProps.iconCategories[1].icons[0];
      expect(wrapper.state().icon).not.toEqual(targetIcon);
      wrapper
        .find('Button')
        .at(1)
        .simulate('click');
      expect(wrapper.state().icon).toEqual(targetIcon);
    });

    test("set state's icon size property on select", () => {
      const targetSize = FeatureStyler.defaultProps.iconSizes[1];
      expect(wrapper.state().iconSize).not.toEqual(targetSize);
      wrapper.find('Select').simulate('change', {
        target: {
          value: targetSize.value.toString(),
        },
      });
      expect(wrapper.state().iconSize).toEqual(targetSize);
    });
  });
});
