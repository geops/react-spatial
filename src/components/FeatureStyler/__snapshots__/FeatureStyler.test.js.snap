// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns  a style object. 1`] = `
<div
  className="tm-feature-styler"
>
  <div
    className="tm-modify-icons"
  >
    <div>
      Modify icon
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/marker.png"
          height={24}
          src="images/marker.png"
          width={24}
        />
      </div>
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/airport.png"
          height={24}
          src="images/airport.png"
          width={24}
        />
      </div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/fuel.png"
          height={24}
          src="images/fuel.png"
          width={24}
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-icon-size"
  >
    <div>
      Modify icon size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="48,48"
    >
      <option
        value={
          Array [
            24,
            24,
          ]
        }
      >
        small size
      </option>
      <option
        value={
          Array [
            36,
            36,
          ]
        }
      >
        medium size
      </option>
      <option
        value={
          Array [
            48,
            48,
          ]
        }
      >
        big size
      </option>
    </select>
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns  an array of 3 styles targeting the one at index 1. 1`] = `
<div
  className="tm-feature-styler"
>
  <div
    className="tm-modify-icons"
  >
    <div>
      Modify icon
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/marker.png"
          height={24}
          src="images/marker.png"
          width={24}
        />
      </div>
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/airport.png"
          height={24}
          src="images/airport.png"
          width={24}
        />
      </div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/fuel.png"
          height={24}
          src="images/fuel.png"
          width={24}
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-icon-size"
  >
    <div>
      Modify icon size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="48,48"
    >
      <option
        value={
          Array [
            24,
            24,
          ]
        }
      >
        small size
      </option>
      <option
        value={
          Array [
            36,
            36,
          ]
        }
      >
        medium size
      </option>
      <option
        value={
          Array [
            48,
            48,
          ]
        }
      >
        big size
      </option>
    </select>
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns  an array of one style. 1`] = `
<div
  className="tm-feature-styler"
>
  <div
    className="tm-modify-icons"
  >
    <div>
      Modify icon
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/marker.png"
          height={24}
          src="images/marker.png"
          width={24}
        />
      </div>
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/airport.png"
          height={24}
          src="images/airport.png"
          width={24}
        />
      </div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/fuel.png"
          height={24}
          src="images/fuel.png"
          width={24}
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-icon-size"
  >
    <div>
      Modify icon size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="48,48"
    >
      <option
        value={
          Array [
            24,
            24,
          ]
        }
      >
        small size
      </option>
      <option
        value={
          Array [
            36,
            36,
          ]
        }
      >
        medium size
      </option>
      <option
        value={
          Array [
            48,
            48,
          ]
        }
      >
        big size
      </option>
    </select>
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns  an empty array. 1`] = `
<div
  className="tm-feature-styler"
/>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns a style with bold font. 1`] = `
<div
  className="tm-feature-styler"
>
  <div>
    <div>
      Modify text
    </div>
    <textarea
      onChange={[Function]}
      rows="1"
      value=""
    />
  </div>
  <div
    className="tm-modify-text-size"
  >
    <div>
      Modify text size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="1"
    >
      <option
        value={1}
      >
        small size
      </option>
      <option
        value={1.5}
      >
        medium size
      </option>
      <option
        value={2}
      >
        big size
      </option>
    </select>
  </div>
  <div
    className="tm-modify-text-font"
  >
    <div
      className="tm-button tm-button-text-bold"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      F
    </div>
  </div>
  <div
    className="tm-modify-text-color"
  >
    <div
      className="tm-color-label"
    >
      Modify color
    </div>
    <div
      className="tm-color-picker"
    >
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          style={
            Object {
              "backgroundColor": "black",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-text-bg-color"
  >
    <div
      className="tm-color-label"
    >
      Modify text background
    </div>
    <div
      className="tm-color-picker"
    >
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          style={
            Object {
              "backgroundColor": "black",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-text-rotation"
  >
    <div>
      Modify text rotation
    </div>
    <input
      max="360"
      min="0"
      onChange={[Function]}
      type="number"
      value="0"
    />
    <div>
      Â°
    </div>
    <input
      max="360"
      min="0"
      onChange={[Function]}
      rows="1"
      type="range"
      value="0"
    />
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns a style with color values selected from properties. 1`] = `
<div
  className="tm-feature-styler"
>
  <div
    className="tm-modify-stroke"
  >
    <div
      className="tm-line-icon-label"
    >
      Modify line closure
    </div>
    <div
      className="tm-line-icon-wrapper"
    >
      <div>
        <div
          className="tm-line-icon-sel"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="tm-line-icon-current"
          >
            <img
              alt="images/noArrow.png"
              src="images/noArrow.png"
            />
          </div>
          <svg
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          className="tm-line-icon-sel"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="tm-line-icon-current"
          >
            <img
              alt="images/noArrow.png"
              src="images/noArrow.png"
            />
          </div>
          <svg
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="tm-modify-color"
  >
    <div
      className="tm-color-label"
    >
      Modify line color
    </div>
    <div
      className="tm-color-picker"
    >
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has a styleFunction() which returns a style with icon and text values selected from properties. 1`] = `
<div
  className="tm-feature-styler"
>
  <div>
    <div>
      Modify text
    </div>
    <textarea
      onChange={[Function]}
      rows="1"
      value="bar"
    />
  </div>
  <div
    className="tm-modify-text-size"
  >
    <div>
      Modify text size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="1.5"
    >
      <option
        value={1}
      >
        small size
      </option>
      <option
        value={1.5}
      >
        medium size
      </option>
      <option
        value={2}
      >
        big size
      </option>
    </select>
  </div>
  <div
    className="tm-modify-text-font"
  >
    <div
      className="tm-button"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      F
    </div>
  </div>
  <div
    className="tm-modify-text-color"
  >
    <div
      className="tm-color-label"
    >
      Modify color
    </div>
    <div
      className="tm-color-picker"
    >
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-text-bg-color"
  >
    <div
      className="tm-color-label"
    >
      Modify text background
    </div>
    <div
      className="tm-color-picker"
    >
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          style={
            Object {
              "backgroundColor": "black",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-text-rotation"
  >
    <div>
      Modify text rotation
    </div>
    <input
      max="360"
      min="0"
      onChange={[Function]}
      type="number"
      value="29"
    />
    <div>
      Â°
    </div>
    <input
      max="360"
      min="0"
      onChange={[Function]}
      rows="1"
      type="range"
      value="29"
    />
  </div>
  <div
    className="tm-modify-icons"
  >
    <div>
      Modify icon
    </div>
    <div>
      <div
        className="tm-button tm-selected"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/marker.png"
          height={24}
          src="images/marker.png"
          width={24}
        />
      </div>
    </div>
    <div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/airport.png"
          height={24}
          src="images/airport.png"
          width={24}
        />
      </div>
      <div
        className="tm-button"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
        tabIndex={0}
      >
        <img
          alt="images/fuel.png"
          height={24}
          src="images/fuel.png"
          width={24}
        />
      </div>
    </div>
  </div>
  <div
    className="tm-modify-icon-size"
  >
    <div>
      Modify icon size
    </div>
    <select
      className="tm-select"
      onChange={[Function]}
      value="24,24"
    >
      <option
        value={
          Array [
            24,
            24,
          ]
        }
      >
        small size
      </option>
      <option
        value={
          Array [
            36,
            36,
          ]
        }
      >
        medium size
      </option>
      <option
        value={
          Array [
            48,
            48,
          ]
        }
      >
        big size
      </option>
    </select>
  </div>
</div>
`;

exports[`FeatureStyler matches snapshot when feature has no styleFunction() defined 1`] = `null`;

exports[`FeatureStyler matches snapshot when no feature provided 1`] = `null`;
